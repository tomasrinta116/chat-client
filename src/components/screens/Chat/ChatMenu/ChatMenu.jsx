import React from "react";
import IconButton from "@material-ui/core/IconButton";
import Menu from "@material-ui/core/Menu";
import MenuItem from "@material-ui/core/MenuItem";
import MoreVertIcon from "@material-ui/icons/MoreVert";
import ParticipantsList from "../ParticipantsList/ParticipantsList";
import { exitChannel } from "../../../../utils/sendbirdHelpers";
import DeleteChannelButton from "../DeleteChannelButton/DeleteChannelButton";
import PropTypes from 'prop-types';

class ChatMenu extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      anchorEl: null,
      isOpen: false
    };
  }

  handleClick = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  handleLeave = () => {
    const { channel, history } = this.props;
    exitChannel(channel);
    history.push("/channels");
  };

  toggleParticipants = () => {
    this.setState({
      isOpen: !this.state.isOpen
    });
    this.handleClose();
  };

  render() {
    const { participants, channel, history, sb } = this.props;
    const { anchorEl, isOpen } = this.state;
    const open = Boolean(anchorEl);

    return (
      <React.Fragment>
        <IconButton data-testid={"chatMenuButton"} onClick={this.handleClick}>
          <MoreVertIcon />
        </IconButton>
        <Menu anchorEl={anchorEl} open={open} onClose={this.handleClose}>
          <MenuItem onClick={this.toggleParticipants}>
            View Participants
          </MenuItem>
          <MenuItem onClick={this.handleLeave}>Leave Channel</MenuItem>
          <DeleteChannelButton history={history} channel={channel} sb={sb} />
        </Menu>
        <ParticipantsList
          isOpen={isOpen}
          toggle={this.toggleParticipants}
          channel={channel}
          participants={participants}
        />
      </React.Fragment>
    );
  }
}

ChatMenu.propTypes = {
  history: PropTypes.object.isRequired,
  channel: PropTypes.object.isRequired,
  participants: PropTypes.arrayOf(PropTypes.string).isRequired,
  sb: PropTypes.object.isRequired,
}

export default ChatMenu;
