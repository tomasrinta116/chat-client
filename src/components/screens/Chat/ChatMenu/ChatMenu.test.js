import React from "react";
import {shallow} from 'enzyme';
import ChatMenu from "./ChatMenu";

describe("<ChatMenu />", () => {
  let wrapper;
  const props = {
    participants: ["test1", "test2", "test3"],
    channel: { isOperatorWithUserId: jest.fn() }, //get this to return true
    sb: { getCurrentUserId: jest.fn() },
    history: {},
  };

  beforeEach(() => {
    wrapper = shallow(<ChatMenu {...props} />);
    })

  test("does it render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

});
