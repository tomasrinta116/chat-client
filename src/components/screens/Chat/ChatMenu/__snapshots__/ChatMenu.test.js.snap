// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMenu /> does it render correctly 1`] = `
<Fragment>
  <WithStyles(IconButton)
    data-testid="chatMenuButton"
    onClick={[Function]}
  >
    <pure(MoreVertIcon) />
  </WithStyles(IconButton)>
  <WithStyles(Menu)
    anchorEl={null}
    onClose={[Function]}
    open={false}
  >
    <WithStyles(MenuItem)
      onClick={[Function]}
    >
      View Participants
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      onClick={[Function]}
    >
      Leave Channel
    </WithStyles(MenuItem)>
    <WithStyles(DeleteChannelButton)
      channel={
        Object {
          "isOperatorWithUserId": [MockFunction],
        }
      }
      history={Object {}}
      sb={
        Object {
          "getCurrentUserId": [MockFunction],
        }
      }
    />
  </WithStyles(Menu)>
  <WithStyles(ParticipantsList)
    channel={
      Object {
        "isOperatorWithUserId": [MockFunction],
      }
    }
    isOpen={false}
    participants={
      Array [
        "test1",
        "test2",
        "test3",
      ]
    }
    toggle={[Function]}
  />
</Fragment>
`;
